app/controllers/dockers_controller.rb

```diff
@@ -0,0 +1,32 @@
+class DockersController < ApplicationController
+  action_auth_level :index, :instructor
+  def index; end
+
+  action_auth_level :uploadDockerImage, :instructor
+  def uploadDockerImage
+    image_name = params[:image_name]
+    if image_name.nil? || image_name.empty?
+      flash[:error] = "Please specify an image name."
```

Comment:
You should probably have an early redirect upon errors? Otherwise if the image file is specified the request is still sent to Tango